src/crypto/crypto_util.cc

```diff
@@ -205,8 +214,13 @@ void GetFipsCrypto(const FunctionCallbackInfo<Value>& args) {
 }
 
 void SetFipsCrypto(const FunctionCallbackInfo<Value>& args) {
+  Mutex::ScopedLock lock(per_process::cli_options_mutex);
+  Mutex::ScopedLock fips_lock(fips_mutex);
+
   CHECK(!per_process::cli_options->force_fips_crypto);
   Environment* env = Environment::GetCurrent(args);
+  // TODO: This should not be possible to set from worker threads.
+  // CHECK(env->owns_process_state());
```

Comment:
Left out because this could be seen as a semver-major change. If we don’t think it’s semver-major, I’m happy to add this as well.

Reply:
Is it worth doing this for Node.js 18 as a semver-major anyway?